src/main/java/com/ebay/feed/api/FeedImpl.java

```diff
@@ -295,6 +298,24 @@ private GetFeedResponse invoker(Request.Builder requestBuilder, Path path, boole
 
         requestBuilder.removeHeader(Constants.RANGE_HEADER);
         requestBuilder.addHeader(Constants.RANGE_HEADER, val);
+        LOGGER.debug("Firing the request for next batch" + newUpperLimit);
+        AuthRequest authRequest = new AuthRequest(credentialFilePath, null);
+
+        // load credentials and generate token
+        CredentialLoader credentialLoader = new CredentialLoader(authRequest);
+        String token = "foobar";
+        try {
+            credentialLoader.loadCredentials();
+            token = credentialLoader.getOauthResponse().getAccessToken().get().getToken();
```

Comment:
Prefer to check with expire time and add logic to get token.

Reply:
ok. will make the required changes.